monash-human-power/dashboard
# Add stories for NavBar in Typscript

## Description

`NavBar` is now a presentational component in Typescript.

Stories have been added for V2 and V3 data.

Converted the `router` module into Typescript as well, though they also depend on `view/v2` components that have not been converted.

## Screenshots

![image](https://user-images.githubusercontent.com/50545432/95657478-64bff600-0b60-11eb-85c0-f59ac7c3a364.png)

![image](https://user-images.githubusercontent.com/50545432/95657472-5eca1500-0b60-11eb-8381-936ca682ad9a.png)

![image](https://user-images.githubusercontent.com/50545432/95657454-4eb23580-0b60-11eb-846d-407d385d49d3.png)

![image](https://user-images.githubusercontent.com/50545432/95657462-5540ad00-0b60-11eb-9c84-598dda7d0926.png)


## Steps to Test

`NavBar` changes

1. Start `client` of `dashboard`

2. It should function the same

Stories

1. Run `yarn run storybook`

2. `NavBar` stories should look the same as on the client

## Problems

![image](https://user-images.githubusercontent.com/50545432/95657416-20ccf100-0b60-11eb-9b47-4cb34ccaac42.png)

I seem to get a bunch of import errors (though it still compiles). Not sure if it's just like that or my config is spazzing out.

The components I imported from `views/v2` for use in `router` have been casted into components. Might want to change that later.
